include/bluetooth/services/bas_c.rst

```diff
@@ -0,0 +1,47 @@
+.. _bas_c_readme:
+
+GATT Battery Service (BAS) Client
+#################################
+
+The Battery Service Client can be used to retrieve information about the battery level from a device that provides a Battery Service Server.
+
+The client supports a simple `Battery Service <Battery Service Specification_>`_ with one characteristic.
+The Battery Level Characteristic holds the battery level in percentage units.
+
+
+Usage
+*****
+
+.. note::
+   Do not access any of the values in the :cpp:type:`bt_gatt_bas_c` object structure directly.
+   All values that should be accessed have accessor functions.
+   The reason that the structure is fully defined is to allow the application to allocate the memory for it.
+
+There are different ways to retrieve the battery level:
+
+Notifications
+  Use :cpp:func:`bt_gatt_bas_c_subscribe` to receive notifications from the connected Battery Service.
+  The notifications are passed to the provided callback function.
+
+  Note that it is not mandatory for the Battery Level Characteristic to support notifications.
+  If the server does not support notifications, read the current value as described below instead.
+
+Reading the current value
+  Use :cpp:func:`bt_gatt_bas_c_read` to read the battery level.
+  When subscribing to notifications, use this function to read the initial value of the battery level after connection.
```

Comment:
Is this correct? The input sounded like you can only get the initial value.

Reply:
```suggestion
  Use this function to read the initial value of the battery level after connection.
  This is also the only access to the battery level in server does not support the notification.
```

Reply:
This is very similar to the initial input, so it doesn't really help me understand, sorry.

bt_gatt_bas_c_read will give me the current value every time I call it, no matter if I use notifications or not - correct?
So if I use notifications, do I need to call bt_gatt_bas_c_read after connection? Or can I rely on notifications only?
And if I don't use notifications, it shouldn't matter if I call bt_gatt_bas_c_read right after connection or later when I actually want to get the latest value?

Reply:
Well... When you wish to relay on notifications only there may be a problem with initial value. In our examples the battery state is reported every second, but in normal use case it is possible that battery state would be reported for example in 10% resolution. And then if the battery can last for 14 days, you may receive a single notification in more than a day interval. So when you are connecting and you do not know what is the battery state, it is good idea to read it.

But yes - you may read it in any time and it would return current value.

Reply:
Thx, now I think I get it. :)
Changed to:
Reading the current value
  Use :cpp:func:`bt_gatt_bas_c_read` to read the battery level.
  When subscribing to notifications, you should call this function every time you connect to retrieve the current value.

Reply:
Well... still not 100% true. As if you are bonded and disconnect, after connection you should get the notification if the value has changed while we were disconnected. :)

Reply:
How about:
When subscribing to notifications, you can call this function to retrieve the current value even when there is no change.
?

Reply:
Looks good.

Reply:
Great :) - updated.

I also changed back the requirements to what you had in your original input. That's consistent to the other samples, let's go for that version. :)

Ready to squash imo. :+1: